src/admin/metrics.rs

```diff
@@ -0,0 +1,336 @@
+use http_body_util::Full;
+use hyper::{
+    body::Bytes,
+    server::conn::http1,
+    service::service_fn,
+    Request,
+};
+use prometheus_metric_storage::{
+    MetricStorage,
+    StorageRegistry,
+};
+use std::convert::Infallible;
+use std::{
+    collections::hash_map::HashMap,
+    sync::{
+        Arc,
+        RwLock,
+    },
+    time::Duration,
+};
+
+use tokio::sync::{
+    mpsc::{
+        unbounded_channel,
+        UnboundedReceiver,
+        UnboundedSender,
+    },
+    oneshot,
+};
+
+use thiserror::Error;
+
+pub type MetricSender = UnboundedSender<RpcMetrics>;
+pub type MetricReceiver = UnboundedReceiver<RpcMetrics>;
+
+// #[cfg(feature = "prometheusd")]
```

Comment:
if everything metrics related is contained within the metrics mod we can conditionally include it at compile time at the mod level